lightning/src/ln/channelmanager.rs

```diff
@@ -3494,27 +3494,35 @@ where
 		let mut peer_state_lock = peer_state_mutex.lock().unwrap();
 		let peer_state = &mut *peer_state_lock;
 		for channel_id in channel_ids {
-			if !peer_state.channel_by_id.contains_key(channel_id) {
+			let context = if let Some(channel) = peer_state.inbound_v1_channel_by_id.get_mut(channel_id) {
+				&mut channel.context
+			} else if let Some(channel) = peer_state.outbound_v1_channel_by_id.get_mut(channel_id) {
+				&mut channel.context
+			} else if let Some(channel) = peer_state.channel_by_id.get_mut(channel_id) {
+				let mut config = channel.context.config();
+				config.apply(config_update);
```

Comment:
Does this actually work? `ChannelContext::config` returns a copy, not a reference. So the change is being applied to a local variable. We may be missing some test coverage if this is the case.

Reply:
Uhhh yeah let me test that. Didn’t change any functionality (just moved) but could be missing that coverage!

Reply:
Oh, yeah, you're right. That's a strange interface but I guess it preexists the PR.